docs/theory/diffusion/diffusion.rst

```diff
@@ -0,0 +1,667 @@
+
+With Pylake, we can calculate the MSD for a :class:`~lumicks.pylake.kymotracker.kymotrack.KymoTrack` with :meth:`~lumicks.pylake.kymotracker.kymotrack.KymoTrack.msd`.
+This returns a tuple of time and MSD values, which we can directly plot::
+
+    for track in tracks:
+        plt.plot(*track.msd())
+
+.. image:: figures/track_msd.png
+
+Note how the curves diverge quickly for larger time lags.
+Larger lags have far fewer points contributing to them.
+MSDs follow a gamma distribution becoming more and more Gaussian with more averaging :cite:`michalet2010mean,michalet2012optimal`.
+This means that smaller lags generally show a symmetric almost Gaussian distribution, while large (less averaged) lags show a much bigger variance.
+By simulating a large number of tracks, we can visualize this::
+
+    tracks = lk.simulation.simulate_diffusive_tracks(diffusion_constant=1.0, steps=12, dt=0.01, observation_noise=0.1, num_tracks=1500)
+    frame_lags = tracks[0].msd()[0]
+    all_msds = np.asarray([track.msd()[1] for track in tracks]).T
+    histogram_edges = np.arange(0, np.max(all_msds), 0.02)
+    msd_histogram = np.array([np.histogram(msd_frame, histogram_edges, density=True)[0] for msd_frame in all_msds])
+    histogram_centers = (histogram_edges[:-1] + histogram_edges[1:]) / 2
+
+    fig = plt.figure(figsize=(10, 5))
+    ax = fig.add_subplot(121, projection='3d')
+    view_limit = 0.8
+    for lag_time, hist in zip(frame_lags, msd_histogram):
+        ax.bar(
+            histogram_centers[histogram_centers < view_limit],
+            hist[histogram_centers < view_limit],
+            zs=lag_time,
+            zdir='y',
+            alpha=0.8,
+            width=histogram_centers[1]-histogram_centers[0]
+        )
+
+    plt.xlim([0, 0.8])
+    ax.view_init(20, -60, 0)
+    ax.set_zticks([])
+    ax.set_xlabel(r'MSD [$\mu m^2$]')
+    ax.set_ylabel('Lag time [s]')
+    plt.tight_layout();
+
+.. image:: figures/msd_distributions.png
+
+If a group of tracks experience the same environment and diffusion coefficient, then it makes sense to compute an averaged MSD estimate using :meth:`~lumicks.pylake.kymotracker.kymotrack.KymoTrackGroup.ensemble_msd`::
+
+    tracks = lk.simulation.simulate_diffusive_tracks(diffusion_constant=1.0, steps=40, dt=0.01, observation_noise=0.1, num_tracks=25)
+    ensemble_msd = tracks.ensemble_msd()
+
+This returns a weighted average of the MSDs coming from all the tracks (where the weight is determined by the number of points that contribute to each estimate).
+If the tracks are of equal length, this weighting will not have an effect (since all the weights will be the same).
+You can plot the ensemble msd as follows::
+
+    ensemble_msd.plot()
+
+.. image:: figures/ensemble_msd.png
+
+MSD estimates should be handled with care.
```

Comment:
This paragraph contains an important conclusion and warning on how to handle MSD. Maybe give it a heading such as 'Summary', or 'Conclusion' or some other synonym of TL;DR

Reply:
The last sentence is optional. I thought it would be nice to list all the experimental artefacts that are accounted for by the various methods in on place.

Reply:
That's a great idea suggestion!

Reading it back, I wonder if `large lags` is descriptive enough since what is large really depends on your track length.

What do you think of something like the following?

```
Important take-aways
^^^^^^^^^^^^^^^^^^^^

MSD estimates should be handled with care.

#. For a densely sampled track, the number of points contributing to a time-averaged MSD estimate is inversely linearly related to the lag time.
#. Poorly averaged MSDs will show a high variance and should not be assumed to be Gaussian distributed.
#. MSDs for different lags are highly correlated and one should not fit these values as though they were independent data points unless care is taken that it is valid to do so (more on this below).

In addition, there are other experimental parameters, such as motion blur and localization accuracy that have to be accounted for.
In the next sections, we will discuss various methods for diffusion analysis that take these variables into account.
```

Reply:
So, the point you want to make in the first item, is that MSD at longer lag times is averaged over fewer data points? And with 'densely sampled track' you mean that there are no gaps and that with gaps, the relationship would not be linear? Is it important to imphasize that? How about:

For larger lag times, MSD values are averaged over fewer lags. This means that the uncertainty of the MSD estimate increases with lag time.

Reply:
Otherwise I like your suggestion!

Reply:
Yes, I will amend it. Thank you! 👍